File: src/main/java/com/gitlab/daring/image/command/parameter/IntParam.java
package com.gitlab.daring.image.command.parameter;

import static java.lang.Integer.parseInt;

public class IntParam extends NumberParam<Integer> {

    public IntParam(String sv, String sp) {
        super(sv, sp);
    }

    public IntParam(String sv) {
        this(sv, "");
    }

    public int posVal(int dv) {
        return v > 0 ? v : dv;
    }

    @Override
    public Integer parseValue(String sv) {
        return parseInt(sv);
    }

    @Override
    public void setNumValue(Number v) {
        setValue(v.intValue());
    }

}

File: src/main/kotlin/com/gitlab/daring/image/command/parameter/IntParam.kt
package com.gitlab.daring.image.command.parameter

class IntParam (sv: String, sp: String = "") : NumberParam<Int>(sv, sp) {

    val fv get() = v.toFloat()
    val dv get() = v.toDouble()

    override fun parseValue(sv: String) = sv.toInt()

    override fun setNumValue(v: Number) {
        value = v.toInt()
    }

    fun posVal(dv: Int) = if (v > 0) v else dv

}